### PR TITLE
[On Hold] API-1836: Write a doc "From identifiers to UUID" for partners

### PR DESCRIPTION
**Pull request by tomglvng**

### CODE REVIEW

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+## What are the impacts?
+
+[Products endpoints](https://api.akeneo.com/api-reference.html#Product) will remain available, even when new API endpoints (same roads with a '-uuid' suffix, for example '/api/rest/v1/products-uuid') will be available.
+Nevertheless, when the current product identifier will turn to optional:
+- `GET /api/rest/v1/products` won’t return products with empty product identifiers (in other words, you may miss products if you continue to use this endpoint);
+- `associations` property for `GET /api/rest/v1/products` or `GET /api/rest/v1/products/{id}` may contain NULL values (product associated with a product without identifier);
+- `GET /api/rest/v1/products/{identifier}` could result in a 404 error (if the identifier is removed from the product).
+...and you could encounter some issues with your code.
+
+## What do we advise?
```
Comment (LevFlavien) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:43]: Also: do not use both old and new endpoints in the same workflow.

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                Update the product in the eCommerce
+            else
+                Add the product to the eCommerce
+                Add (uuid, ecommerce_identifier) in the correlation table
+        else
+            Update the product identified by ecommerce_identifier in the eCommerce
+  ```
+
+::: info
+If you need help, don't hesitate to contact us!
```
Comment (meganelepalud-akeneo) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:103]:
```suggestion
If you need help, don't hesitate to [contact us](https://www.akeneo.com/contact/)!
```

**Review by meganelepalud-akeneo — APPROVED** [review]

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+- `PATCH /api/rest/v1/products-uuid`, same as [PATCH /api/rest/v1/products](https://api.akeneo.com/api-reference.html#patch_products)
+- `GET /api/rest/v1/products-uuid/{uuid}`, same as [GET /api/rest/v1/products/{code}](https://api.akeneo.com/api-reference.html#get_products__code_)
+- `PATCH /api/rest/v1/products-uuid/{uuid}`, same as [PATCH /api/rest/v1/products/{code}](https://api.akeneo.com/api-reference.html#patch_products__code_)
+- `DELETE /api/rest/v1/products-uuid/{uuid}`, same as [DELETE /api/rest/v1/products/{code}](https://api.akeneo.com/api-reference.html#delete_products__code_)
+
+And later (during the last quarter of 2022), we will make optional the product identifier (`pim_catalog_identifier` attribute).
+
+## Why do we do that?
+
+At the time these lines are written (July 2022), a PIM product contains one and only one way of unique identification: the so-called property `identifier` (the only `pim_catalog_identifier` attribute of the whole product).
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:25]:
```suggestion
At the time these lines are written (July 2022), a PIM product contains one and only one way of unique identification: the so-called field `identifier` (the only `pim_catalog_identifier` attribute of the whole product).
In Sererity, this attribute always called 'SKU'
```

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+Usually, this field value is simply the SKU (Stock Keeping Unit) of the product, but what if you need to identify your product with several product identifiers (SKU, EAN, GTIN,...)?
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:26]: How to identify a product if SKU change ?

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+Adding classic fields won't do the job: you need a kind of identifier field for each product identifier.
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:27]:
```suggestion
Adding classic fields won't do the job: you need a kind of identifier field for each product.
```

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+That's the purpose of the brand-new multi-identifier feature.
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:29]:
```suggestion
That's the purpose of the brand-new product UUID feature.
```

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+But before making it happen, a product must have a **unique** and **immutable** way to identify it: that's why we introduce the product UUID (for Universally Unique identifier).
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:31]:
```suggestion
But before making it happen, a product must have a **unique** and **immutable** way to identify it: that's why we introduce the product UUID (for Universally Unique Identifier).
```

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+- `associations` property for [GET /api/rest/v1/products](https://api.akeneo.com/api-reference.html#get_products) or [GET /api/rest/v1/products/{code}](https://api.akeneo.com/api-reference.html#get_products__code_) may contain **NULL** values (product associated with a product without identifier);
+-  [GET /api/rest/v1/products/{code}](https://api.akeneo.com/api-reference.html#get_products__code_) could result in a 404 error (if the identifier is removed from the product).
+  ...and you could encounter some issues with your code.
+
+## What do we advise you?
+
+As soon as product UUID API endpoints are released, jump on the bandwagon right away:
+- Change [product-identifier-based products endpoints](https://api.akeneo.com/api-reference.html#Product) with UUID ones.
+- Every time you use a product identifier in your application, be sure to replace it by corresponding product UUID. You will easily find the match between product identifier and UUID in both [products endpoints](https://api.akeneo.com/api-reference.html#Product) and new ones.
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:46]:
```suggestion
- Every time you identify a product with its identifier in your application, be sure to replace it by corresponding product UUID. You will easily find the match between product identifier and UUID in both [products endpoints](https://api.akeneo.com/api-reference.html#Product) and new ones.
```

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+:::warning
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:48]: To remove

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+Usually, this field value is simply the SKU (Stock Keeping Unit) of the product, but what if you need to identify your product with several product identifiers (SKU, EAN, GTIN,...)?
+Adding classic fields won't do the job: you need a kind of identifier field for each product identifier.
+
+That's the purpose of the brand-new multi-identifier feature.
+
+But before making it happen, a product must have a **unique** and **immutable** way to identify it: that's why we introduce the product UUID (for Universally Unique identifier).
+
+## What are the impacts?
+
+Of course, [Products endpoints](https://api.akeneo.com/api-reference.html#Product) will remain available, even when new API endpoints will be ready for use.
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:35]: Old endpoints will have a uuid property

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+:::warning
+Don't mix API endpoints in the same workflow, or you may encounter some inconsistencies!
+:::
+
+## An example of what you may have to do?
+
+Let's imagine you developed a product-identifier-based solution that synchronizes PIM products to a random eCommerce solution.
+
+Its running is as follows:
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:56]:
```suggestion
Today, its running is as follows:
```

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+    if ecommerce_identifier was not found
+      Add the product to the eCommerce
+      Add (identifier, ecommerce_identifier) in the correlation table
+    else
+      Update the product identified by ecommerce_identifier in the eCommerce
+  ```
+
+The identifiers' correlation table looks like:
+
+```code
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:72]: Before algorithm

File: content/getting-started/from-identifiers-to-uuid-7x/welcome.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+    TABLE correlation {
+      akeneo_identifier,
+      ecommerce_identifier
+  }
+  ```
+
+How do you turn your application into a product-UUID one?
+
+First, replace the way of getting Akeneo products: from `GET /api/rest/v1/products` to `GET /api/rest/v1/products-uuid`.
```
Comment (tomglvng) [inline content/getting-started/from-identifiers-to-uuid-7x/welcome.md:81]: Add a column in correlation 'akeneo_uuid'

**Review by pierallard — APPROVED** [review]
GTM seen IRL some fixes

**Pull request merged**